### DISCUSSION TITLE
Why setting init_state variable as`init_value` in SCIF model

### DISCUSSION THREAD

**Discussion by nnbtam** [lava-nc/lava-optimization#230] [Q&A] [answered]
Hi Lava team,

I'm new to the neuromorphic computing paradigm and I have a quick question regarding the implementation of SCIF model [https://github.com/lava-nc/lava-optimization/blob/74e58b5f361eac79cea51b1d748529863d4c66b7/src/lava/lib/optimization/solvers/generic/sub_process_models.py#LL69C1-L69C1](here)

I just wonder why we set the default value of `init_state` variable as `init_value` for SCIF neuron model.

Thank you very much in advance!
Tam

**Comment by AlessandroPierro** [comment, marked as answer]
Hi Tam, thanks for reaching out! 

The link you provided is not working for me, but your question is very relevant. Looking at the code [here](https://github.com/lava-nc/lava-optimization/blob/74e58b5f361eac79cea51b1d748529863d4c66b7/src/lava/lib/optimization/solvers/generic/sub_process_models.py#L69), I can confirm we are wrongly initializing the `init_state` variable with the value passed for `init_value`. 

If you want to start contributing to lava-optimisation, please make a Pull Request to fix the bug. I will do the review and approve it. We would really appreciate your contribution! 🎉 

In any case, I will take care of fixing the bug before the next release.


Thank you for helping us improve Lava-Optimisation for the whole community!

-- Alessandro
